frontend/dispatcher.lua

```diff
@@ -414,14 +415,19 @@ function Dispatcher:addItem(caller, menu, location, settings, section)
                     end,
                     callback = function(touchmenu_instance)
                         local SpinWidget = require("ui/widget/spinwidget")
+                        local requested_precision = settingsList[k].precision
+                        if settingsList[k].step and math.floor(settingsList[k].step) ~= settingsList[k].step then
+                            requested_precision = "%0.1f"
+                        end
```

Comment:
That's slightly funkier and doesn't change the fact that it stomps on the class defaults if it's unset in `settingsList` (or if step is unset, or if step is an integer) :?

Reply:
Yes, the fact that we use the defaults in numberpicker doesn't change. 
But the need of entering a precision string in the settings is dropped, which makes the settingsList shorter.

Reply:
AFAICT, that still breaks NumberPicker because you risk stomping its `precision` default with a `nil`?